substrate/frame/assets/src/migration/v1.rs

```diff
@@ -0,0 +1,172 @@
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::{AssetStatus, Config, DepositBalanceOf, Pallet, LOG_TARGET};
+use frame_support::{
+	pallet_prelude::*, sp_runtime::traits::Saturating, traits::OnRuntimeUpgrade, weights::Weight,
+};
+use log;
+
+#[cfg(feature = "try-runtime")]
+use crate::Vec;
+#[cfg(feature = "try-runtime")]
+use sp_runtime::TryRuntimeError;
+
+mod old {
+	use super::*;
+
+	#[derive(Decode)]
+	pub struct AssetDetails<Balance, AccountId, DepositBalance> {
+		pub owner: AccountId,
+		pub issuer: AccountId,
+		pub admin: AccountId,
+		pub freezer: AccountId,
+		pub supply: Balance,
+		pub deposit: DepositBalance,
+		pub min_balance: Balance,
+		pub is_sufficient: bool,
+		pub accounts: u32,
+		pub sufficients: u32,
+		pub approvals: u32,
+		pub is_frozen: bool,
+	}
+
+	impl<Balance, AccountId, DepositBalance> AssetDetails<Balance, AccountId, DepositBalance> {
+		pub(super) fn migrate_to_v1(
+			self,
+		) -> super::AssetDetails<Balance, AccountId, DepositBalance> {
+			let status = if self.is_frozen { AssetStatus::Frozen } else { AssetStatus::Live };
+
+			super::AssetDetails {
+				owner: self.owner,
+				issuer: self.issuer,
+				admin: self.admin,
+				freezer: self.freezer,
+				supply: self.supply,
+				deposit: self.deposit,
+				min_balance: self.min_balance,
+				is_sufficient: self.is_sufficient,
+				accounts: self.accounts,
+				sufficients: self.sufficients,
+				approvals: self.approvals,
+				status,
+			}
+		}
+	}
+}
+
+#[frame_support::storage_alias]
+/// Details of an asset.
+pub(crate) type Asset<T: Config<I>, I: 'static> = StorageMap<
+	Pallet<T, I>,
+	Blake2_128Concat,
+	<T as Config<I>>::AssetId,
+	AssetDetails<
+		<T as Config<I>>::Balance,
+		<T as frame_system::Config>::AccountId,
+		DepositBalanceOf<T, I>,
+	>,
+>;
+
+#[derive(PartialEq, Eq, Clone, Encode, Decode, RuntimeDebug, TypeInfo)]
+pub(crate) struct AssetDetails<Balance, AccountId, DepositBalance> {
+	pub owner: AccountId,
+	pub issuer: AccountId,
+	pub admin: AccountId,
+	pub freezer: AccountId,
+	pub supply: Balance,
+	pub deposit: DepositBalance,
+	pub min_balance: Balance,
+	pub is_sufficient: bool,
+	pub accounts: u32,
+	pub sufficients: u32,
+	pub approvals: u32,
+	pub status: AssetStatus,
+}
+
+pub struct MigrateToV1<T, I>(core::marker::PhantomData<(T, I)>);
+impl<T: Config<I>, I: 'static> OnRuntimeUpgrade for MigrateToV1<T, I> {
```

Comment:
Can use `VersionedMigration` for this and avoid doing version checks/sets yourself